### PR TITLE
Add sanity/existence check for `kubecostModel.federatedStorageConfigSecret`

### LINKED ISSUE #3074: [Feature] Perform Secret existence check on `kubecostModel.federatedStorageConfigSecret`
### Problem Statement

A sanity/existence check for `kubecostProductConfigs.cloudIntegrationSecret` was added in PR #3048, but a similar problem could result by not specifying `kubecostModel.federatedStorageConfigSecret` in that the chart will proceed with installation only to fail when coming up if the referenced Secret is not available. We can apply the same UX improvement to this value as we did to cloudIntegrationSecret.

### Solution Description

Perform a sanity/existence check for `kubecostModel.federatedStorageConfigSecret` and fail if either the Secret cannot be found or does not have the expected key.

### Alternatives

_No response_

### Additional Context

See PR #3048 as well as follow-up in #3055

### Troubleshooting

- [X] I have read and followed the [issue guidelines](https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/ISSUE_GUIDELINES.md) and this is a feature request only for the Helm chart.
- [X] I have searched other issues in this repository and mine is not recorded.

### PR DESCRIPTION
**Pull request by chipzoller**
Signed-off-by: chipzoller <chipzoller@gmail.com>

## What does this PR change?

Adds a sanity/existence check for the Secret defined at `kubecostModel.federatedStorageConfigSecret` similar to what PR #3048 did for `kubecostProductConfigs.cloudIntegrationSecret`.

## Does this PR rely on any other PRs?

No

## How does this PR impact users? (This is the kind of thing that goes in release notes!)

Provides a better UX in that the chart installation will fail if the Secret cannot be found.

## Links to Issues or tickets this PR addresses or fixes

Closes #3074

## What risks are associated with merging this PR? What is required to fully test this PR?

Secret check could be incorrect.

## How was this PR tested?

Tested locally with `helm install`.

## Have you made an update to documentation? If so, please provide the corresponding PR.

N/A

### CODE REVIEW

File: cost-analyzer/values.yaml [hunk 1]
```diff
@@ -512,8 +512,9 @@ kubecostModel:
 
   # The name of the Secret containing a bucket config for ETL backup.
   # etlBucketConfigSecret:
-  # The name of the Secret containing a bucket config for Federated storage.
-  # federatedStorageConfigSecret:
+  # The name of the Secret containing a bucket config for Federated storage. The contents should be stored
+  # under a key named federated-store.yaml.
+  federatedStorageConfigSecret: ""
```
Comment (michaelmdresser) [inline cost-analyzer/values.yaml]: Why uncomment this? I think we only need it for certain configurations

Comment (chipzoller) [reply cost-analyzer/values.yaml]: Mistake and fixed! I appreciate you, Michael.

**Review by JesseGoodier — APPROVED** [review]

**Comment by santinoncs** [comment]
What happens when you install using helm template ?

I am using argoCD to deploy the helm chart and natively argoCD uses  

```
helm template ..
```

And when this happens, the check is considered failed.

**Comment by chipzoller** [comment]
See #3083.

`helm template` works. Argo CD doesn't support Helm's `lookup` function. Please put some pressure on them to support.

**Comment by michaelmdresser** [comment]
@santinoncs if you need the check to be skipped, consider using `.Values.global.platforms.cicd.skipSanityChecks`. Have a look at this PR to understand what it does.

**Comment by santinoncs** [comment]
> @santinoncs if you need the check to be skipped, consider using `.Values.global.platforms.cicd.skipSanityChecks`. Have a look at this PR to understand what it does.

true! sorry i was checking the v2.0 version

**Pull request merged**